src/app/core/store/content/viewcontexts/viewcontexts.reducer.ts

```diff
@@ -0,0 +1,55 @@
+import { EntityState, createEntityAdapter } from '@ngrx/entity';
+import { createReducer, on } from '@ngrx/store';
+
+import { CallParameters } from 'ish-core/models/call-parameters/call-parameters.model';
+import { ContentPageletEntryPoint } from 'ish-core/models/content-pagelet-entry-point/content-pagelet-entry-point.model';
+import { setLoadingOn } from 'ish-core/utils/ngrx-creators';
+
+import {
+  loadViewContextEntrypoint,
+  loadViewContextEntrypointFail,
+  loadViewContextEntrypointSuccess,
+} from './viewcontexts.actions';
+
+declare type ContentPageletEntryPointWithContext = ContentPageletEntryPoint & {
+  viewContextId: string;
+  callParameters: CallParameters;
+};
+
+export function serializeContextSpecificViewContextId(viewContextId: string, callParameters: CallParameters) {
+  return (
+    viewContextId +
+    Object.keys(callParameters)
+      .map(key => `@${key}-${callParameters[key]}`)
```

Comment:
As you need both key and value it might be better to use `Object.entires`.
On a different note: it might be also good to sort the keys, so view contexts are unique independent of call parameters order.
\+ add test for this 😉